code-challenges.test.js

```diff
@@ -34,19 +62,81 @@ const letterA = "a"
+    it("takes in an array of words and a single letter and returns an array of all the words containing that particular letter.", () => {
+        expect(containsLetter(fruitArray, "a")).toEqual(["Mango", "Apricot", "Peach"])
+        expect(containsLetter(fruitArray, "e")).toEqual(["Cherry", "Blueberry", "Peach"])
+    })
+})
+//     ReferenceError: containsLetter is not defined
+
 // b) Create the function that makes the test pass.
+// create a function called containsLetter that takes in an array and a string (which should contain a single letter)
+// iterate through each word in the array
+// for each word, use the built in string method .includes() to check if that word contains the given letter
+// the iterator should be a filter since we are removing items from the array
+
+//     - Expected  - 1
+// + Received  + 0
+
+// Array [
+//   "Mango",
+// -   "Apricot",
+//   "Peach",
+// ]
+// forgot to account for capitalization again! I will lowercase the value before using includes
+
+const containsLetter = (array, string) => {
+return array.filter(value => value.toLowerCase().includes(string))
+}
+//Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
-// Expected output: true
+describe("fullhouse", () => {
+    const hand1 = [5, 5, 5, 3, 3]
+    const hand2 = [5, 5, 3, 3, 4]
+    const hand3 = [5, 5, 5, 5, 4]
+    const hand4 = [7, 2, 7, 2, 7]
+    it("takes in an array of 5 numbers and determines whether or not the array is a full house", () => {
+        expect(fullhouse(hand1)).toEqual(true)
+        expect(fullhouse(hand2)).toEqual(false)
+        expect(fullhouse(hand3)).toEqual(false)
+        expect(fullhouse(hand4)).toEqual(true)
+    })
+})
+
+//     ReferenceError: fullhouse is not defined
 
 // b) Create the function that makes the test pass.
+// create a function called fullhouse that takes in an array as parameters
+// create two variables which will be placeholders 
+// iterate over the array assign the first two unique numbers to the placeholders. on each iteration it will check if it is equal to either of the placeholders and if so, assign the value to the placeholder name and add 1 to the count for that placeholder. if not, do nothing. 
+// I did find in my research a method called countBy that I thought could be useful, but it only applied to objects, not arrays, and seemed to involve installing various JS librarues such as underscore, collect.js, or lodash
+const fullhouse = (array) => {
```

Comment:
This looks excellent!  Excited to see that you are utilizing higher-order functions.  All of your code in this section is really efficient and looks great.  Awesome work!  👏